test/parallel/test-tls-close-event-after-write.js

```diff
@@ -12,21 +12,22 @@ const tls = require('tls');
 const fixtures = require('../common/fixtures');
 let cconn = null;
 let sconn = null;
+let read_len = 0;
+const buffer_size = 1024 * 1024;
 
 function test() {
   if (cconn && sconn) {
     cconn.resume();
     sconn.resume();
-    sconn.end(Buffer.alloc(1024 * 1024));
-    cconn.end();
+    sconn.end(Buffer.alloc(buffer_size));
   }
 }
 
 const server = tls.createServer({
   key: fixtures.readKey('agent1-key.pem'),
   cert: fixtures.readKey('agent1-cert.pem')
 }, function(c) {
-  c.on('close', function() {
+  c.once('close', function() {
```

Comment:
This only fires once, right? In that case I have a mild preference for using `.on()` and testing that the callback is only called once, i.e.

```suggestion
  c.on('close', common.mustCall(() => server.close()));
```

Reply:
Updated my change with your suggestion.